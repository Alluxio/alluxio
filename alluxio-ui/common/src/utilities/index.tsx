/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export * from './saga/getSagaRequest';

export * from './misc/getDebouncedFunction';
export * from './misc/parseQueryString';
export * from './misc/isExternalLink';
export * from './misc/bytesToString';
export * from './misc/disableFormSubmit';
export * from './misc/renderFileNameLink';

export * from './nivo/transformToNivoFormat';
