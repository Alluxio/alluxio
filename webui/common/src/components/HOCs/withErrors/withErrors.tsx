/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import {Alert} from "reactstrap";
import {IAlertErrors} from "../../../constants";
import {getDisplayName} from "../../../utilities";

export interface IErrorProps {
    errors: IAlertErrors;
}

export function withErrors<T extends IErrorProps>(WrappedComponent: React.ComponentType<T>) {
    const errorsHoc = (props: T) => {
        const {errors} = props;

        return errors.hasErrors
            ? (
                <Alert color="danger">
                    {errors.general && <div>Unable to reach the api endpoint for this page.</div>}
                    {errors.specific.map((err, i) => <div key={i}>{err}</div>)}
                </Alert>
            )
            : <WrappedComponent {...props} />
    };
    (errorsHoc as React.FunctionComponent).displayName = `withErrors(${getDisplayName(WrappedComponent)})`;
    return errorsHoc;
}
