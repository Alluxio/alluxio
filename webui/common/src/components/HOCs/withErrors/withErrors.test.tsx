/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {configure, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {IErrorProps, withErrors} from "./withErrors";
import {createAlertErrors} from "../../../utilities";

configure({adapter: new Adapter()});

const WrappedComponent = () => <div>Wrapped</div>;
const EnhancedComponent = withErrors(WrappedComponent);

describe('withErrors HOC', () => {
    let props: IErrorProps;

    beforeAll(() => {
        props = {
            errors: createAlertErrors(false)
        };
    });

    describe('Shallow component', () => {
        let shallowWrapper: ShallowWrapper;

        beforeAll(() => {
            shallowWrapper = shallow(<EnhancedComponent {...props}/>);
        });

        it('Renders without crashing', () => {
            expect(shallowWrapper.length).toEqual(1);
        });

        describe('No Errors', () => {
            it('Matches snapshot - renders WrappedComponent', () => {
                shallowWrapper.setProps({ errors: createAlertErrors(false) });
                expect(shallowWrapper).toMatchSnapshot();
            });
        });

        describe('With Errors', () => {
            it('Matches snapshot - renders general error', () => {
                shallowWrapper.setProps({ errors: createAlertErrors(true) });
                expect(shallowWrapper).toMatchSnapshot();
            });

            it('Matches snapshot - renders specific errors', () => {
                shallowWrapper.setProps({ errors: createAlertErrors(true, ['error1', 'error2']) });
                expect(shallowWrapper).toMatchSnapshot();
            });
        });
    });
});
