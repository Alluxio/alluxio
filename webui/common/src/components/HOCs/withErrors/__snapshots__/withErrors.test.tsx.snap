// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withErrors HOC Shallow component No Errors Matches snapshot - renders WrappedComponent 1`] = `
<WrappedComponent
  errors={
    Object {
      "general": false,
      "hasErrors": false,
      "specific": Array [],
    }
  }
/>
`;

exports[`withErrors HOC Shallow component With Errors Matches snapshot - renders general error 1`] = `
<Alert
  closeAriaLabel="Close"
  color="danger"
  fade={true}
  isOpen={true}
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  <div>
    Unable to reach the api endpoint for this page.
  </div>
</Alert>
`;

exports[`withErrors HOC Shallow component With Errors Matches snapshot - renders specific errors 1`] = `
<Alert
  closeAriaLabel="Close"
  color="danger"
  fade={true}
  isOpen={true}
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  <div>
    Unable to reach the api endpoint for this page.
  </div>
  <div
    key="0"
  >
    error1
  </div>
  <div
    key="1"
  >
    error2
  </div>
</Alert>
`;
